watchtower/standalone.go

```diff
@@ -142,6 +152,39 @@ func (w *Standalone) Stop() error {
 	return nil
 }
 
+// createNewHiddenService automatically sets up a v2 or v3 onion service in
+// order to listen for inbound connections over Tor.
+func (w *Standalone) createNewHiddenService() error {
+	// Get all the ports the watchtower is listening on. These will be used to
+	// map the hidden service's virtual port.
+	listenPorts := make([]int, 0, len(w.listeners))
+	for _, listener := range w.listeners {
+		port := listener.Addr().(*net.TCPAddr).Port
+		listenPorts = append(listenPorts, port)
+	}
+
+	// Once we've created the port mapping, we can automatically create the
+	// hidden service. The service's private key will be saved on disk in order
+	// to persistently have access to this hidden service across restarts.
+	onionCfg := tor.AddOnionConfig{
+		VirtualPort: DefaultPeerPort,
+		TargetPorts: listenPorts,
+		Store:       tor.NewOnionFile(w.cfg.WatchtowerKeyPath, 0600),
+		Type:        w.cfg.Type,
+	}
+
+	addr, err := w.cfg.TorController.AddOnion(onionCfg)
+	if err != nil {
+		return err
+	}
+
+	// Append this address to ExternalIPs so that it will be exposed in
+	// tower info calls.
+	w.cfg.ExternalIPs = append(w.cfg.ExternalIPs, addr)
```

Comment:
maybe we should make a private `externalIPs` field so that we don't modify the config?

Reply:
It seems that this config is restricted to only `standalone.go` so I don't think it matters if we append to the config. What was your thinking?

Reply:
Yes the manner in which it is integrated in lnd it is fine to do so. From an API perspective though i think it's better not to manipulate configs directly since you can't make assumptions about how/if the caller is using those values elsewhere. Wouldn't say it's a blocking issue tho.

Reply:
I think we can move on without the private `externalIPs` field. The command `lncli tower info` eventually calls `func (w *Standalone) ExternalIPs() []net.Addr` which is just a getter for `w.cfg.ExternalIPs` to display the correct "uris" in the output. That means we have to modify `w.cfg.ExternalIPs` to have an onion address displayed since those aren't passed in via config and are instead created. But the config is also private (`cfg *Config`) so an API caller would also call `ExternalIPs()` to access the config field. Which means we would need two getters, one for the private field and one for the config field.